### PR TITLE
Closes #443 | Add dataloader for ASR-STIDUSC

### LINKED ISSUE #443: Create dataset loader for ASR-STIDUSC: A SCRIPTED THAI DAILY-USE SPEECH CORPUS
Dataloader name: `asr_stidusc/asr_stidusc.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?asr_stidusc

|     Dataset| asr_stidusc |
|-------------|---|
| Description | This open-source dataset consists of 4.56 hours of transcribed Thai scripted speech focusing on daily use sentences, where 5,431 utterances contributed by ten speakers were contained. |
| Subsets | - |
| Languages | tha |
| Tasks | Text-To-Speech Synthesis, Automatic Speech Recognition |
| License | Creative Commons Attribution Non Commercial No Derivatives 4.0 (cc-by-nc-nd-4.0) |
| Homepage | https://magichub.com/datasets/thai-scripted-speech-corpus-daily-use-sentence/ |
| HF URL | - |
| Paper URL | - |

### PR DESCRIPTION
**Pull request by zwenyu**
Closes #443 

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/{my_dataset}/{my_dataset}.py` (please use only lowercase and underscore for dataset folder naming, as mentioned in dataset issue) and its `__init__.py` within `{my_dataset}` folder.
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_LOCAL`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py` or `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py --subset_id {subset_name_without_source_or_seacrowd_suffix}`.
- [x] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

File: seacrowd/sea_datasets/asr_stidusc/asr_stidusc.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+"""
+
+_HOMEPAGE = "https://magichub.com/datasets/thai-scripted-speech-corpus-daily-use-sentence/"
+_LANGUAGES = ["tha"]
+_LICENSE = Licenses.CC_BY_NC_ND_4_0.value
+_LOCAL = False
+_URLS = {
+    _DATASETNAME: "https://magichub.com/df/df.php?file_name=Thai_Scripted_Speech_Corpus_Daily_Use_Sentence.zip",
+}
+_SUPPORTED_TASKS = [Tasks.TEXT_TO_SPEECH, Tasks.SPEECH_RECOGNITION]
```
Comment (elyanah-aco) [inline seacrowd/sea_datasets/asr_stidusc/asr_stidusc.py:43]:
```suggestion
_SUPPORTED_TASKS = [Tasks.SPEECH_RECOGNITION]
```

File: seacrowd/sea_datasets/asr_stidusc/asr_stidusc.py [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+_SOURCE_VERSION = "1.0.0"
+_SEACROWD_VERSION = "1.0.0"
+
+
+class ASRStidusc(datasets.GeneratorBasedBuilder):
+    """ASR-Stidusc consists transcribed Thai scripted speech focusing on daily use sentences"""
```
Comment (elyanah-aco) [inline seacrowd/sea_datasets/asr_stidusc/asr_stidusc.py:50]:
```suggestion
    """ASR-STIDuSC consists transcribed Thai scripted speech focusing on daily use sentences"""
```

File: seacrowd/sea_datasets/asr_stidusc/asr_stidusc.py [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+_URLS = {
+    _DATASETNAME: "https://magichub.com/df/df.php?file_name=Thai_Scripted_Speech_Corpus_Daily_Use_Sentence.zip",
+}
+_SUPPORTED_TASKS = [Tasks.TEXT_TO_SPEECH, Tasks.SPEECH_RECOGNITION]
+
+_SOURCE_VERSION = "1.0.0"
+_SEACROWD_VERSION = "1.0.0"
+
+
+class ASRStidusc(datasets.GeneratorBasedBuilder):
```
Comment (elyanah-aco) [inline seacrowd/sea_datasets/asr_stidusc/asr_stidusc.py:49]:
```suggestion
class ASRSTIDuSCDataset(datasets.GeneratorBasedBuilder):
```

File: seacrowd/sea_datasets/asr_stidusc/asr_stidusc.py [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+            elif s[2] == "F":
+                s[2] = "female"
+            else:
+                s[2] = None
+        # dictionary of metadata of each speaker
+        spkinfo_dict = {s[1]: {"speaker_gender": s[2], "speaker_age": int(s[3]), "speaker_region": s[4], "speaker_device": s[5]} for s in spkinfo_data}
+
+        num_sample = len(uttransinfo_data)
+
+        for i in range(num_sample):
```
Comment (elyanah-aco) [inline seacrowd/sea_datasets/asr_stidusc/asr_stidusc.py:153]: This is more Pythonic:
```suggestion
        for i, sample in enumerate(uttransinfo_data):
```
Then you can replace all instances of `uttransinfo_data[i]` in the for loop to `sample`.

**Review by elyanah-aco — CHANGES_REQUESTED** [review]
Hi @zwenyu, thanks for the PR! Dataloader works well on my end, just have some nits:

**Comment by zwenyu** [comment]
@elyanah-aco Thanks for the comments. I've pushed changes according to your suggestions

**Review by ljvmiranda921 — APPROVED** [review]
LGTM!

**Review by elyanah-aco — APPROVED** [review]
Thanks, LGTM! Merging in a bit.

**Pull request merged**